File: mac_gui/send_to_windows/Sources/TransferCore/Log.swift
// Log.swift — §4.7 phase-1 shim 로그 (plain text 3줄).
//
// 형식 (Windows shim 출력 미러):
//   [ts] context-menu send: <abs-src> -> <abs-dst>
//   [ts] mode=<file|directory|batch>  hash=<full-sha>  payload=<n> bytes
//   [ts] state=ready transfer_id=<transfer-id>
//
// 같은 ts를 3줄에 모두 적용 (호출 시점 기준).

import Foundation

public enum TransferLog {

    public static func render(transferID: String,
                              mode: TransferMode,
                              sourceAbsPath: String,
                              destAbsPath: String,
                              hashHex: String,
                              payloadBytes: Int64,
                              at: Date = .init(),
                              timeZone: TimeZone = .current,
                              entryKind: String = "context-menu send") -> String {
        let ts = "[\(Timestamps.logTimestamp(at, timeZone: timeZone))]"
        return """
        \(ts) \(entryKind): \(sourceAbsPath) -> \(destAbsPath)
        \(ts) mode=\(mode.rawValue)  hash=\(hashHex)  payload=\(payloadBytes) bytes
        \(ts) state=ready transfer_id=\(transferID)

        """
    }
}

File: mac_gui/share-manager/src-tauri/src/transfer/log.rs
// log.rs — port of Log.swift. §4.7 phase-1 shim plain-text log (3 lines).
//
//   [ts] context-menu send: <abs-src> -> <abs-dst>
//   [ts] mode=<file|directory|batch>  hash=<full-sha>  payload=<n> bytes
//   [ts] state=ready transfer_id=<transfer-id>

use chrono::{DateTime, Local};

use super::timestamps;

pub fn render(
    transfer_id: &str,
    mode: &str,
    source_abs: &str,
    dest_abs: &str,
    hash_hex: &str,
    payload_bytes: u64,
    at: DateTime<Local>,
    entry_kind: &str,
) -> String {
    let ts = format!("[{}]", timestamps::log_timestamp(at));
    format!(
        "{ts} {entry_kind}: {source_abs} -> {dest_abs}\n\
         {ts} mode={mode}  hash={hash_hex}  payload={payload_bytes} bytes\n\
         {ts} state=ready transfer_id={transfer_id}\n"
    )
}
